libs/zzz/formula/src/data/disc/disc.test.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  compileTagMapValues,
+  setDebugMode,
+} from '@genshin-optimizer/pando/engine'
+import type {
+  DiscMainStatKey,
+  DiscSetKey,
+  DiscSubStatKey,
+} from '@genshin-optimizer/zzz/consts'
+import { data, keys, values } from '..'
+import {
+  charTagMapNodeEntries,
+  conditionals,
+  discTagMapNodeEntries,
+  teamData,
+  withMember,
+} from '../..'
+import { Calculator } from '../../calculator'
+import type { Read } from '../util'
+import {
+  conditionalEntries,
+  convert,
+  enemy,
+  enemyDebuff,
+  own,
+  ownBuff,
+  ownTag,
+  type TagMapNodeEntries,
+} from '../util'
+
+setDebugMode(true)
+// This is generally unnecessary, but without it, some tags in `DebugCalculator` will be missing
+Object.assign(values, compileTagMapValues(keys, data))
+
+function testCharacterData(
+  setKey: DiscSetKey,
+  discStats: Partial<Record<DiscMainStatKey | DiscSubStatKey, number>> = {},
+  otherCharData: TagMapNodeEntries = []
+) {
+  const data: TagMapNodeEntries = [
+    ...teamData(['Anby']),
+    ...withMember(
+      'Anby',
+      ...charTagMapNodeEntries({
+        level: 60,
+        promotion: 5,
+        key: 'Anby',
+        mindscape: 0,
+        basic: 0,
+        dodge: 0,
+        special: 0,
+        assist: 0,
+        chain: 0,
+        core: 6,
+      }),
+      ...discTagMapNodeEntries(discStats, {
+        [setKey]: 4,
+      }),
+      ...otherCharData
+    ),
+    own.common.critMode.add('avg'),
+    enemy.common.def.add(953),
+    enemy.common.res_.electric.add(0.1),
+    enemy.common.isStunned.add(0),
+    enemyDebuff.common.resRed_.electric.add(0.15),
+    enemyDebuff.common.dmgInc_.add(0.1),
+    enemyDebuff.common.dmgRed_.add(0.15),
+    enemyDebuff.common.stun_.add(1.5),
+  ]
+  return data
+}
+function cond(setKey: DiscSetKey, name: string, value: number) {
+  return conditionalEntries(setKey, 'Anby', 'Anby')(name, value)
+}
+function printDebug(calc: Calculator, read: Read) {
+  console.log(JSON.stringify(calc.toDebug().compute(read), undefined, 2))
```

Comment:
merge with master and you can change to 
```suggestion
  console.log(prettify(calc.toDebug().compute(read)))
```
and filtering will also be added automatically